moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp

```diff
@@ -484,77 +277,54 @@ bool LMAKinematicsPlugin::searchPositionIK(const geometry_msgs::Pose& ik_pose, c
                                     << ik_pose.position.x << " " << ik_pose.position.y << " " << ik_pose.position.z
                                     << " " << ik_pose.orientation.x << " " << ik_pose.orientation.y << " "
                                     << ik_pose.orientation.z << " " << ik_pose.orientation.w);
-  // Do the IK
-  for (unsigned int i = 0; i < dimension_; i++)
-    jnt_seed_state(i) = ik_seed_state[i];
-  jnt_pos_in = jnt_seed_state;
-
-  unsigned int counter(0);
-  while (1)
+  unsigned int attempt = 0;
+  do
   {
-    //    ROS_DEBUG_NAMED("lma","Iteration: %d, time: %f, Timeout:
-    //    %f",counter,(ros::WallTime::now()-n1).toSec(),timeout);
-    counter++;
-    if (timedOut(n1, timeout))
+    ++attempt;
+    if (attempt > 1)  // randomly re-seed after first attempt
     {
-      ROS_DEBUG_NAMED("lma", "IK timed out");
-      error_code.val = error_code.TIMED_OUT;
-      ik_solver_vel.unlockRedundantJoints();
-      return false;
+      if (!consistency_limits.empty())
+        getRandomConfiguration(jnt_seed_state.data, consistency_limits, jnt_pos_in.data);
+      else
+        getRandomConfiguration(jnt_pos_in.data);
+      ROS_DEBUG_STREAM_NAMED("lma", "New random configuration (" << attempt << "): " << jnt_pos_in);
     }
+
     int ik_valid = ik_solver_pos.CartToJnt(jnt_pos_in, pose_desired, jnt_pos_out);
-    ROS_DEBUG_NAMED("lma", "IK valid: %d", ik_valid);
-    if (!consistency_limits.empty())
+    if (ik_valid == 0 || options.return_approximate_solution)  // found acceptable solution
     {
-      getRandomConfiguration(jnt_seed_state, consistency_limits, jnt_pos_in, options.lock_redundant_joints);
-      if ((ik_valid < 0 && !options.return_approximate_solution) ||
-          !checkConsistency(jnt_seed_state, consistency_limits, jnt_pos_out))
-      {
-        ROS_DEBUG_NAMED("lma", "Could not find IK solution: does not match consistency limits");
+      if (!consistency_limits.empty() && !checkConsistency(jnt_seed_state.data, consistency_limits, jnt_pos_out.data))
```

Comment:
It's unfortunate that we can't pass the consistency_limits into the solver

Reply:
We can't.